### PR TITLE
Provide reason for dropped txs

### PR DESCRIPTION
**Pull request by danlaine**

### CODE REVIEW

File: vms/platformvm/service.go [hunk 1]
```diff
@@ -1911,34 +1912,79 @@ func (service *Service) GetTx(_ *http.Request, args *api.GetTxArgs, response *ap
...
+	IncludeReason bool `json:"includeReason"`
+}
+
+// GetTxStatusResponse ...
+type GetTxStatusResponse struct {
+	includeReason bool
+	Status        `json:",omitempty"`
+	// Reason this tx was dropped.
+	// Only non-empty if Status is dropped
+	Reason string `json:"reason"`
```
Comment (StephenButtolph) [inline vms/platformvm/service.go:1940]: These shouldn't have JSON annotations... right?

File: vms/platformvm/service.go [hunk 2]
```diff
@@ -1911,34 +1912,79 @@ func (service *Service) GetTx(_ *http.Request, args *api.GetTxArgs, response *ap
...
+}
+
+func (r GetTxStatusResponse) MarshalJSON() ([]byte, error) {
+	if !r.includeReason {
+		return stdjson.Marshal(r.Status)
+	}
+	if r.Reason != "" {
+		return []byte(fmt.Sprintf("{\"status\": \"%s\", \"reason\": \"%s\"}", r.Status, r.Reason)), nil
+	}
+	return []byte(fmt.Sprintf("{\"status\": \"%s\"}", r.Status)), nil
```
Comment (StephenButtolph) [inline vms/platformvm/service.go:1950]: I think this should just return `\"%s\", r.Status` right?

File: vms/platformvm/service.go (hunk 1 above)
Comment (danlaine) [reply vms/platformvm/service.go]: fixed

File: vms/platformvm/service.go (hunk 2 above)
Comment (danlaine) [reply vms/platformvm/service.go]: fixed

**Pull request merged**